website/frontend/src/components/pages/Programs/children/Message/Message.tsx

```diff
@@ -53,16 +56,30 @@ export const Message: VFC<Props> = ({ programId, programName, handleClose }) =>
 
   useEffect(() => {
     if (!meta) {
-      getMeta()
-        .then((res) => {
-          setMeta(JSON.parse(res.result.meta) ?? null);
-        })
-        .catch((err: RPCResponseError) => dispatch(AddAlert({ type: EventTypes.ERROR, message: err.message })))
-        .finally(() => {
-          setReady(true);
-        });
+      if (chain === 'Development') {
+        localPrograms
+          .getItem(programId)
+          .then((response: any) => {
+            setMeta(JSON.parse(response.meta.meta) ?? null);
+          })
+          .catch((error) => {
+            dispatch(AddAlert({ type: EventTypes.ERROR, message: `Error: ${error}` }));
+          })
+          .finally(() => {
+            setReady(true);
+          });
+      } else {
+        getMeta()
+          .then((res) => {
+            setMeta(JSON.parse(res.result.meta) ?? null);
+          })
+          .catch((err: RPCResponseError) => dispatch(AddAlert({ type: EventTypes.ERROR, message: err.message })))
+          .finally(() => {
+            setReady(true);
+          });
+      }
```

Comment:
```suggestion
const fetchMeta = useCallback(async (id: string) => {
    const apiRequest = new ServerRPCRequestService();

    return apiRequest.callRPC<GetMetaResponse>(RPC_METHODS.GET_METADATA, { id });
  }, []);

  const getMeta = chain === 'Development' ? localPrograms.getItem : fetchMeta;

  useEffect(() => {
    if (!meta) {
      getMeta(programId)
        .then((res) => {
          setMeta(JSON.parse(res.result.meta) ?? null);
        })
        .catch((err: RPCResponseError) => dispatch(AddAlert({ type: EventTypes.ERROR, message: err.message })))
        .finally(() => {
          setReady(true);
        });
    }
  }, [meta, programId, getMeta, dispatch]);
```
I suggest to think about one general variable that would choose what function you need, and then handle it's execution the same way regardless of its choice, though localPrograms.getItem should return the same structure as fetchMeta.
I think it's much clearer solution that probably can be applied to other components too,

Reply:
Yes, it's a very good idea 👍 but as for me the method **getItem** should returns the structure like our usual Program structure. However, If we really want to change it, guess, it's another task.